### PR TITLE
Add createPattern, setPatternFillStyle to Graphics.Canvas

### PR DESCRIPTION
**Pull request by igagen**
This adds the following to the public Graphics.Canvas API:

* `createPattern` - bindings to the canvas function for creating repeatable patterns from images.
* `setPatternFillStyle` - sets the current fill style to a previously created pattern.
* `PatternRepeat` - enumerates the different types of pattern repetition (used in `createPattern`).
* `CanvasPattern` - opaque data type returned by `createPattern`.

### CODE REVIEW

File: src/Graphics/Canvas.purs [hunk 1]
```diff
@@ -471,6 +479,21 @@ foreign import drawImageScale :: forall eff. Context2D -> CanvasImageSource -> N
...
 foreign import drawImageFull :: forall eff. Context2D -> CanvasImageSource -> Number -> Number -> Number -> Number -> Number -> Number -> Number -> Number -> Eff (canvas :: Canvas | eff) Context2D
 
+-- | Enumerates the different types of pattern repetitions.
+data PatternRepeat = Repeat | RepeatX | RepeatY | NoRepeat
+
+foreign import createPatternImpl :: forall eff. CanvasImageSource -> String -> Context2D -> Eff (canvas :: Canvas | eff) CanvasPattern
+
+-- | Create a new canvas pattern (repeatable image).
+createPattern :: forall eff. CanvasImageSource -> PatternRepeat -> Context2D -> Eff (canvas :: Canvas | eff) CanvasPattern
+createPattern img Repeat   = createPatternImpl img "repeat"
```
Comment (paf31) [inline src/Graphics/Canvas.purs]: I think we should do something consistent with other enum strings like `TextAlign`. I don't particularly like the TextAlign solution of using `Show`, so I suggest both data types get changed to have a (valid) `Show` instance, and a helper function to convert to their string values.

But that can be another commit. This looks good to merge.

Thanks!

Comment (igagen) [inline src/Graphics/Canvas.purs]: Thanks for merging this. I can add the `Show` instance for `PatternRepeat`. For the helper, what would you call that, `toString`?

Comment (paf31) [inline src/Graphics/Canvas.purs]: Sounds good.
